lib/ansible/module_utils/basic.py

```diff
@@ -1611,7 +1611,12 @@ def _check_type_float(self, value):
 
     def _check_type_path(self, value):
         value = self._check_type_str(value)
-        return os.path.expanduser(os.path.expandvars(value))
+        valid_path = os.path.expanduser(os.path.expandvars(value))
+
+        if valid_path == '':
```

Comment:
If an empty string is invalid path then it should be fixed for all modules.

Any of the os.path functions used in this case return `None` so no need to test it. If `None` is passed to any of the functions, the function rises `TypeError: argument of type 'NoneType' is not iterable` error.